### ISSUE TITLE
Last activation not quantized

### ISSUE THREAD

**Issue by L-ED** [pytorch/pytorch#87592]
### 🐛 Describe the bug

I'm trying to quantize resnet-based model:

ResNets(
  (input_stem): ModuleList(
    (0): ConvLayer(
      (conv): Conv2d(3, 32, kernel_size=(3, 3), stride=(2, 2), padding=(1, 1), bias=False)
      (bn): BatchNorm2d(32, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (act): ReLU(inplace=True)
    )
    (1): ConvLayer(
      (conv): Conv2d(32, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
      (bn): BatchNorm2d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (act): ReLU(inplace=True)
    )
  )
  (max_pooling): MaxPool2d(kernel_size=3, stride=2, padding=1, dilation=1, ceil_mode=False)
  (blocks): ModuleList(
    (0): ResNetBottleneckBlock(
      (conv1): Sequential(
        (conv): Conv2d(64, 48, kernel_size=(1, 1), stride=(1, 1), bias=False)
        (bn): BatchNorm2d(48, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (act): ReLU(inplace=True)
      )
      (conv2): Sequential(
        (conv): Conv2d(48, 48, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
        (bn): BatchNorm2d(48, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (act): ReLU(inplace=True)
      )
      (conv3): Sequential(
        (conv): Conv2d(48, 256, kernel_size=(1, 1), stride=(1, 1), bias=False)
        (bn): BatchNorm2d(256, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      )
      (downsample): Sequential(
        (ident): IdentityLayer()
        (conv): Conv2d(64, 256, kernel_size=(1, 1), stride=(1, 1), bias=False)
        (bn): BatchNorm2d(256, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      )
      (final_act): ReLU(inplace=True)
    )
    (1): ResNetBottleneckBlock(
      (conv1): Sequential(
        (conv): Conv2d(256, 88, kernel_size=(1, 1), stride=(1, 1), bias=False)
        (bn): BatchNorm2d(88, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (act): ReLU(inplace=True)
      )
      (conv2): Sequential(
        (conv): Conv2d(88, 88, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
        (bn): BatchNorm2d(88, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (act): ReLU(inplace=True)
      )
      (conv3): Sequential(
        (conv): Conv2d(88, 256, kernel_size=(1, 1), stride=(1, 1), bias=False)
        (bn): BatchNorm2d(256, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      )
      (downsample): IdentityLayer()
      (final_act): ReLU(inplace=True)
    )
   ... SAME BLOCKS
    (14): ResNetBottleneckBlock(
      (conv1): Sequential(
        (conv): Conv2d(1024, 408, kernel_size=(1, 1), stride=(1, 1), bias=False)
        (bn): BatchNorm2d(408, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (act): ReLU(inplace=True)
      )
      (conv2): Sequential(
        (conv): Conv2d(408, 408, kernel_size=(3, 3), stride=(2, 2), padding=(1, 1), bias=False)
        (bn): BatchNorm2d(408, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (act): ReLU(inplace=True)
      )
      (conv3): Sequential(
        (conv): Conv2d(408, 2048, kernel_size=(1, 1), stride=(1, 1), bias=False)
        (bn): BatchNorm2d(2048, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      )
      (downsample): Sequential(
        (avg_pool): AvgPool2d(kernel_size=2, stride=2, padding=0)
        (conv): Conv2d(1024, 2048, kernel_size=(1, 1), stride=(1, 1), bias=False)
        (bn): BatchNorm2d(2048, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      )
      (final_act): ReLU(inplace=True)
    )
    (15): ResNetBottleneckBlock(
      (conv1): Sequential(
        (conv): Conv2d(2048, 512, kernel_size=(1, 1), stride=(1, 1), bias=False)
        (bn): BatchNorm2d(512, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (act): ReLU(inplace=True)
      )
      (conv2): Sequential(
        (conv): Conv2d(512, 512, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
        (bn): BatchNorm2d(512, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
        (act): ReLU(inplace=True)
      )
      (conv3): Sequential(
        (conv): Conv2d(512, 2048, kernel_size=(1, 1), stride=(1, 1), bias=False)
        (bn): BatchNorm2d(2048, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      )
      (downsample): IdentityLayer()
      (final_act): ReLU(inplace=True)
    )
  )
  (global_avg_pool): AdaptiveAvgPool2d(output_size=(1, 1))
  (classifier): LinearLayer(
    (dropout): Dropout(p=0.1, inplace=True)
    (linear): Linear(in_features=2048, out_features=4, bias=True)
  )
)

When i quantize model using fx quantization and all blocks quantized properly except last:

GraphModule(
  (input_stem): Module(
    (0): Module(
      (conv): QuantizedConvReLU2d(3, 32, kernel_size=(3, 3), stride=(2, 2), scale=1.0, zero_point=0, padding=(1, 1))
    )
    (1): Module(
      (conv): QuantizedConvReLU2d(32, 64, kernel_size=(3, 3), stride=(1, 1), scale=1.0, zero_point=0, padding=(1, 1))
    )
  )
  (max_pooling): MaxPool2d(kernel_size=3, stride=2, padding=1, dilation=1, ceil_mode=False)
  (blocks): Module(
    (0): Module(
      (downsample): Module(
        (conv): QuantizedConv2d(64, 256, kernel_size=(1, 1), stride=(1, 1), scale=1.0, zero_point=0)
      )
      (conv1): Module(
        (conv): QuantizedConvReLU2d(64, 48, kernel_size=(1, 1), stride=(1, 1), scale=1.0, zero_point=0)
      )
      (conv2): Module(
        (conv): QuantizedConvReLU2d(48, 48, kernel_size=(3, 3), stride=(1, 1), scale=1.0, zero_point=0, padding=(1, 1))
      )
      (conv3): Module(
        (conv): QuantizedConv2d(48, 256, kernel_size=(1, 1), stride=(1, 1), scale=1.0, zero_point=0)
      )
    )
    (1): Module(
      (conv1): Module(
        (conv): QuantizedConvReLU2d(256, 88, kernel_size=(1, 1), stride=(1, 1), scale=1.0, zero_point=0)
      )
      (conv2): Module(
        (conv): QuantizedConvReLU2d(88, 88, kernel_size=(3, 3), stride=(1, 1), scale=1.0, zero_point=0, padding=(1, 1))
      )
      (conv3): Module(
        (conv): QuantizedConv2d(88, 256, kernel_size=(1, 1), stride=(1, 1), scale=1.0, zero_point=0)
      )
    )
    (2): Module(
      (conv1): Module(
        (conv): QuantizedConvReLU2d(256, 48, kernel_size=(1, 1), stride=(1, 1), scale=1.0, zero_point=0)
      )
      (conv2): Module(
        (conv): QuantizedConvReLU2d(48, 48, kernel_size=(3, 3), stride=(1, 1), scale=1.0, zero_point=0, padding=(1, 1))
      )
      (conv3): Module(
        (conv): QuantizedConv2d(48, 256, kernel_size=(1, 1), stride=(1, 1), scale=1.0, zero_point=0)
      )
    )
    ...
    (14): Module(
      (downsample): Module(
        (avg_pool): AvgPool2d(kernel_size=2, stride=2, padding=0)
        (conv): QuantizedConv2d(1024, 2048, kernel_size=(1, 1), stride=(1, 1), scale=1.0, zero_point=0)
      )
      (conv1): Module(
        (conv): QuantizedConvReLU2d(1024, 408, kernel_size=(1, 1), stride=(1, 1), scale=1.0, zero_point=0)
      )
      (conv2): Module(
        (conv): QuantizedConvReLU2d(408, 408, kernel_size=(3, 3), stride=(2, 2), scale=1.0, zero_point=0, padding=(1, 1))
      )
      (conv3): Module(
        (conv): QuantizedConv2d(408, 2048, kernel_size=(1, 1), stride=(1, 1), scale=1.0, zero_point=0)
      )
    )
    (15): Module(
      (conv1): Module(
        (conv): QuantizedConvReLU2d(2048, 512, kernel_size=(1, 1), stride=(1, 1), scale=1.0, zero_point=0)
      )
      (conv2): Module(
        (conv): QuantizedConvReLU2d(512, 512, kernel_size=(3, 3), stride=(1, 1), scale=1.0, zero_point=0, padding=(1, 1))
      )
      (conv3): Module(
        (conv): QuantizedConv2d(512, 2048, kernel_size=(1, 1), stride=(1, 1), scale=1.0, zero_point=0)
      )
      (final_act): ReLU(inplace=True)  < --- This is problem
    )
  )
  (global_avg_pool): AdaptiveAvgPool2d(output_size=(1, 1))
  (classifier): Module(
    (dropout): QuantizedDropout(p=0.1, inplace=True)
    (linear): QuantizedLinear(in_features=2048, out_features=4, scale=1.0, zero_point=0, qscheme=torch.per_channel_affine)
  )
)

I also checked graph and got dequantization inserts after 14 block:

![Screenshot from 2022-10-24 12-36-55](https://user-images.githubusercontent.com/58368102/197496580-049b3af3-54d1-463b-b5bb-3c3cbc8d5e43.png)

### Versions

Collecting environment information...
PyTorch version: 1.12.1
Is debug build: False
CUDA used to build PyTorch: 11.3
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.6 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.17

Python version: 3.7.13 (default, Mar 29 2022, 02:18:16)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-4.15.0-177-generic-x86_64-with-debian-buster-sid
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: 
GPU models and configuration: 
GPU 0: NVIDIA GeForce RTX 2080 Ti

Nvidia driver version: 510.73.05
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] efficientnet-pytorch==0.7.1
[pip3] numpy==1.21.5
[pip3] segmentation-models-pytorch==0.3.0
[pip3] torch==1.12.1
[pip3] torch-tb-profiler==0.4.0
[pip3] torchtext==0.13.1
[pip3] torchtyping==0.1.4
[pip3] torchvision==0.13.1
[conda] blas                      1.0                         mkl  
[conda] cudatoolkit               11.3.1               ha36c431_9    nvidia
[conda] efficientnet-pytorch      0.7.1                    pypi_0    pypi
[conda] ffmpeg                    4.3                  hf484d3e_0    pytorch
[conda] mkl                       2021.4.0           h06a4308_640  
[conda] mkl-service               2.4.0            py37h7f8727e_0  
[conda] mkl_fft                   1.3.1            py37hd3c417c_0  
[conda] mkl_random                1.2.2            py37h51133e4_0  
[conda] numpy                     1.21.5           py37he7a7128_2  
[conda] numpy-base                1.21.5           py37hf524024_2  
[conda] pytorch                   1.12.1          py3.7_cuda11.3_cudnn8.3.2_0    pytorch
[conda] pytorch-mutex             1.0                        cuda    pytorch
[conda] segmentation-models-pytorch 0.3.0                    pypi_0    pypi
[conda] torch-tb-profiler         0.4.0                    pypi_0    pypi
[conda] torchtext                 0.13.1                     py37    pytorch
[conda] torchtyping               0.1.4                    pypi_0    pypi
[conda] torchvision               0.13.1               py37_cu113    pytorch

**Comment by L-ED** [comment]
I also checked prepared by prepare_qat_fx() model. Last add - relu coupling doesn't have post activation fake-quantization simulation like in previous couplings:

Penultimate add-relu coupling: 
![Screenshot from 2022-10-25 14-15-02](https://user-images.githubusercontent.com/58368102/197759903-8c22feb4-bc80-469d-a725-5bb0e6cf0ea4.png)

Final coupling:
![Screenshot from 2022-10-25 14-15-56](https://user-images.githubusercontent.com/58368102/197760269-69c06580-5917-421b-b95f-392f9df46bce.png)

**Comment by L-ED** [comment]
I fixed problem by replacing x.view(x.size(0), -1) to torch.flatten(x, start_dim=1)
